pkg/server/api_metrics_test.go

```diff
@@ -0,0 +1,267 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatch/cloudwatchiface"
+	"github.com/grafana/grafana/pkg/tsdb"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch"
+	_ "github.com/grafana/grafana/pkg/tsdb/cloudwatch"
+
+	cwapi "github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/grafana/grafana/pkg/api/dtos"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/infra/fs"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrations"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/ini.v1"
+	"xorm.io/xorm"
+)
+
+func createGrafDir(t *testing.T) (string, string) {
```

Comment:
@papagian Yeah, I expect there will be more files, but my thinking is it makes sense to refactor when adding more tests, since it's only then you will know what's needed. See the YAGNI (You Ain't Gonna Need It) principle :)

I'm also wondering if the test should be renamed a bit, but I'm unsure to what. I named it in order to correlate with the tested API endpoint, and lack familiarity with the semantics of the endpoint. Do you know if the semantics of this endpoint is to _only_ query CloudWatch? I can't really tell.

Reply:
I renamed the test to `TestQueryCloudWatchMetrics`, since at least for now it's 100% for CloudWatch.

Reply:
> Do you know if the semantics of this endpoint is to only query CloudWatch? I can't really tell.

no it's not only for cloudwatch. So the aim is to test the endpoint not. specifically cloudwatch, ok then.

Reply:
@papagian It depends a bit on how we want to do it. Maybe there should be separate tests for different uses of the endpoint, or maybe they should be sub-tests. Not sure.